karma.conf.js

```diff
@@ -63,174 +39,68 @@ const SAUCE_LAUNCHERS = {
 const PROGRESS = USE_SAUCE ? 'dots' : 'progress';
-
 const customLaunchers = Object.assign({}, USE_SAUCE ? SAUCE_LAUNCHERS : {}, HEADLESS_LAUNCHERS);
 const browsers = USE_SAUCE ? Object.keys(customLaunchers) : ['Chrome'];
-const istanbulInstrumenterLoader = {
-  use: [{
-    loader: 'istanbul-instrumenter-loader',
-    options: {esModules: true},
-  }],
-  exclude: [
-    /node_modules/,
-    /adapter.[jt]s$/,
-    /animation\/.*$/,
-    /auto-init\/.*$/,
-    /base\/.*$/,
-    /checkbox\/.*$/,
-    /chips\/.*$/,
-    /constants.[jt]s$/,
-    /data-table\/.*$/,
-    /dialog\/.*$/,
-    /dom\/.*$/,
-    /drawer\/.*$/,
-    /floating-label\/.*$/,
-    /form-field\/.*$/,
-    /icon-button\/.*$/,
-    /line-ripple\/.*$/,
-    /linear-progress\/.*$/,
-    /list\/.*$/,
-    /menu\/.*$/,
-    /menu-surface\/.*$/,
-    /notched-outline\/.*$/,
-    /radio\/.*$/,
-    /ripple\/.*$/,
-    /select\/.*$/,
-    /slider\/.*$/,
-    /snackbar\/.*$/,
-    /switch\/.*$/,
-    /tab-bar\/.*$/,
-    /tab-indicator\/.*$/,
-    /tab-scroller\/.*$/,
-    /tab\/.*$/,
-    /textfield\/.*$/,
-    /top-app-bar\/.*$/,
-  ],
-  include: path.resolve('./packages'),
-};
-
-const mochaConfig = {
-  basePath: '',
-  // Refer https://github.com/karma-runner/karma-mocha
-  frameworks: ['mocha'],
-  files: [
-    // Refer https://github.com/babel/karma-babel-preprocessor
-    'node_modules/@babel/polyfill/dist/polyfill.js',
-    'test/unit/index.js',
-  ],
-  preprocessors: {
-    'test/unit/index.js': ['webpack', 'sourcemap'],
-  },
-  reporters: [PROGRESS, 'coverage-istanbul'],
-
-  coverageIstanbulReporter: {
-    'dir': 'coverage',
-    'reports': ['html', 'lcovonly', 'json'],
-    'report-config': {
-      lcovonly: {subdir: '.'},
-      json: {subdir: '.', file: 'coverage.json'},
-    },
-    // Set 'emitWarning = true' to NOT fail tests if the thresholds are not met
-    'emitWarning': false,
-    'thresholds': {
-      statements: 95,
-      branches: 92,
-      lines: 95,
-      functions: 95,
-    },
-  },
 
-  client: {
-    mocha: {
-      reporter: 'html',
-      ui: 'qunit',
-
-      // Number of milliseconds to wait for an individual `test(...)` function to complete.
-      // The default is 2000.
-      timeout: 10000,
-    },
-  },
+// Files to include in Jasmine tests.
+const FILES_TO_USE = [
+  'packages/*/!(node_modules)/**/!(*.d).ts',
+  'packages/*/!(*.d).ts',
+  'testing/**/*.ts',
+];
 
-  webpackMiddleware: {
-    noInfo: true,
-    stats: 'minimal',
-  },
-};
+// Files to exclude in Jasmine tests.
+const EXCLUDE_FILES = [
+  'packages/**/*.scss.test.ts',
+  'testing/featuretargeting/**',
+  'testing/ts-node.register.js',
+  'scripts/**/*.ts',
+];
 
-const jasmineConfig = {
-  basePath: '',
-  files: FILES_TO_USE,
-  exclude: EXCLUDE_FILES,
-  frameworks: ['jasmine', 'karma-typescript'],
-  karmaTypescriptConfig: {
+module.exports = function(config) {
+  config.set({
+    // Jasmine/TS config.
+    // - jasmine: https://github.com/karma-runner/karma-jasmine
```

Comment:
Use direct link to Karma configuration documentation:
http://karma-runner.github.io/4.0/config/configuration-file.html